### PR TITLE
Alternative Frontend Jam

### PR DESCRIPTION
**Pull request by alx-alexpark**
Thanks for making a PR! Please fill in the below:

Name: Alex Park

Do you run a club? No, but I am *maybe* an active community member.

What relevant experiences do you have to help you write this Jam?
I've worked with alternative frontends before, and have experience with python web scraping and generating webpages using python.

Will you be available on Slack to provide support to Hack Clubbers who run into issues while going through your Jam?
Yes! Of course!

### CODE REVIEW

File: jams/making_alternative_frontend.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+**_Platform Limitations_**: It works for all platforms with a browser, or you can run python locally.
+
+**_Time estimation_**: 1-2 hours~
+
+**_Skill level_**: Intermediate-Advanced
+
+### Breakdown
+- Setup: Setting up a development environment, most likely to be in the cloud (Codespaces, Replit, etc...) (5 mins)
+- Part 1: Choosing a website to scrape (10 mins)
```
Comment (zeyuyaoy) [inline jams/making_alternative_frontend.md:20]: I love how this is open-ended, but maybe we can provide some websites that are easy to scrape for those who are trying out this jam as a challenge

File: jams/making_alternative_frontend.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- Part 4: Displaying the extracted content in your custom frontend (30-40 mins)
+
+### Questions:
+1. How will you ensure that every outcome of the workshop varies (i.e., how will you give jammers a sense of ownership over their project)?
+- Everybody will scrape different websites, and will display different content/data on their website's frontend.
+2. Why should Hack Clubbers care about this project?
+- Websites nowdays are constantly getting more and more annoying, not to mention the widespread use of invasive trackers, and bloat. With this jam, high schoolers will be able to learn how to overcome these difficulties by designing their own frontend for popular websites. Building alternative frontends generally makes the internet a better place by improving privacy and overall making the website more pleasant to use. For example, an alternative frontend could greatly simplify a website, strip trackers and ads, and make is much lighter. 
+3. What Club Members will walk away with (both in terms of knowledge and in terms of product)
+- They will walk away with the knowledge to scrape websites, and display that scraped data in their own unique way.
```
Comment (zeyuyaoy) [inline jams/making_alternative_frontend.md:31]: I think you can also have a secondary objective to explain how these Python web frameworks work and how the data flows from the original site and into their app

**Review by cytronicoder — COMMENTED** [review]
Added a few comments for you to think about, but otherwise this looks good!

File: jams/making_alternative_frontend.md (hunk 1 above)
Comment (linkai101) [reply jams/making_alternative_frontend.md:20]: I guess the one concern about this is that some websites might not allow or have barriers to web scraping (CAPCHAs, etc.), so +1 that

**Review by linkai101 — APPROVED** [review]
Love this idea! I could imagine there could be a lot of discussion on this since the new changes to APIs (Twitter, Reddit, etc.) and the relevance of web/data scraping. You could also mention that one of the use cases is for gathering data for statistics research.

I honestly don't think alternative frontends are super practical at least for long-term use cases, since websites are constantly changing and are adding more barriers to stuff like this, but I think it could definitely make for a great jam.

**Comment by sarthaktexas** [comment]
howdy alex! it looks like we have a workshop already similar to this (albeit using diff technologies), so we won't be accepting this, but thank you so much for the proposal!

**Pull request closed without merging**